### DISCUSSION TITLE
More than one glsp in one application?

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#933] [Q&A] [answered]
Hi,
    I have an application that use glsp integrating into theia, Everything is ok if I use one glsp to deal with one kind of file. just like the workflow demo did. Now my situation is that I will deal with two kinds of files. these two kinds of files have no relationship. kind one is just like workflow demo, and kind two does not need edge, So I am thing whether it is possible to create two glsp groups to deal with them. project tree should like 
v1-glsp
v1-server
v1-theia
v2-glsp
v2-server
v2-theia
I am not sure this solution is practicable, Any suggestions?

**Comment by tortmayr** [comment, marked as answer]
Hi @wss29,
yes this solution is definitely feasible. 

In theory, GLSP has a mechanism so that one server can handle multiple diagram types so it might be possible to combine your two server instances into one. However, we have had some reports recently that this feature is not working as expected and might be bugged (https://github.com/eclipse-glsp/glsp/discussions/902 and https://github.com/eclipse-glsp/glsp/issues/804).

So for now, the solution that you outlined is probably the best/safest approach.

**Reply by wss29** [reply]
Hi,
     I encounter one bug when using two glsp
![image](https://user-images.githubusercontent.com/4537809/223647096-e29b4098-9338-4566-8c9e-91dc9885a34d.png)
as you can see the menu has two same items, and warning log is as follow
![image](https://user-images.githubusercontent.com/4537809/223647249-d4c0d792-b13b-422e-a5a9-4aeea4a55083.png)

**Reply by tortmayr** [reply]
Thanks for reporting this, 
I added to problem to this issue: https://github.com/eclipse-glsp/glsp/issues/804.
We will try to provide a sophisticated fix for this soon. In the meantime you should be able to workaround using the configuation flags of the [`GLSPTheiaFrontendModule`](https://github.com/eclipse-glsp/glsp-theia-integration/blob/master/packages/theia-integration/src/browser/glsp-theia-container-module.ts#L45).

With setting the  `enableLayoutCommands` property to false in the module of your second language you should be able to bypass the double registration.

**Reply by wss29** [reply]
hi,
   thanks, problem solved
